contracts/src/PriceFeeds/FXPriceFeed.sol

```diff
@@ -1,50 +1,144 @@
+    * @notice Initializes the FXPriceFeed contract
+    * @param _oracleAdapterAddress The address of the OracleAdapter contract
+    * @param _rateFeedID The address of the rate feed ID
+    * @param _borrowerOperationsAddress The address of the BorrowerOperations contract
+    * @param _watchdogAddress The address of the watchdog contract
+    * @param _initialOwner The address of the initial owner
+    */
+    function initialize(
+        address _oracleAdapterAddress,
+        address _rateFeedID,
+        address _borrowerOperationsAddress,
+        address _watchdogAddress,
+        address _initialOwner
+    ) external initializer {
+        require(_oracleAdapterAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_rateFeedID != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_borrowerOperationsAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_watchdogAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_initialOwner != address(0), "FXPriceFeed: ZERO_ADDRESS");
+
         oracleAdapter = IOracleAdapter(_oracleAdapterAddress);
         rateFeedID = _rateFeedID;
         borrowerOperations = IBorrowerOperations(_borrowerOperationsAddress);
         watchdogAddress = _watchdogAddress;
+
+        _transferOwnership(_initialOwner);
+    }
+
+    /**
+    * @notice Sets the watchdog address
+    * @param _watchdogAddress The address of the new watchdog contract
+    */
+    function setWatchdogAddress(address _watchdogAddress) external onlyOwner {
+        require(_watchdogAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+
+        address oldWatchdogAddress = watchdogAddress;
+        watchdogAddress = _watchdogAddress;
+
+        emit WatchdogAddressUpdated(oldWatchdogAddress, _watchdogAddress);
     }
 
-    function fetchPrice() public returns (uint256, bool) {
+    /**
+    * @notice Fetches the price of the FX rate, if valid
+    * @dev If the contract is shutdown due to oracle failure, the last valid price is returned
+    * @return The price of the FX rate
+    */
+    function fetchPrice() public returns (uint256) {
         if (isShutdown) {
-            return (lastGoodPrice, false);
+            return lastValidPrice;
         }
 
-        (uint256 numerator, ) = oracleAdapter.getFXRateIfValid(rateFeedID);
-
-        lastGoodPrice = numerator;
+        (uint256 price, ) = oracleAdapter.getFXRateIfValid(rateFeedID);
 
-        return (numerator, false);
-    }
+        lastValidPrice = price;
 
-    function fetchRedemptionPrice() external returns (uint256, bool) {
-        return fetchPrice();
+        return price;
     }
 
+    /**
+     * @notice Shuts down the price feed contract due to oracle failure
+     * @dev Can only be called by the authorized watchdog address.
+     *      Once shutdown:
+     *      - The contract will only return the last valid price
+     *      - The BorrowerOperations contract is notified to shut down the collateral branch
```

Comment:
nit: I feel like it's worth adding that trove manager is also notified via borrower ops.